gammapy/detect/lima.py

```diff
@@ -0,0 +1,129 @@
+        Background map
+    kernel : `astropy.convolution.Kernel2D`
+        convolution kernel. 
+    exposure : `~numpy.ndarray`
+        Exposure map
+    fft : bool
+        Use fast fft convolution. Default is False.
+    
+    Returns
+    -------
+    Bunch : `gammapy.extern.bunch.Bunch`
+        Bunch of result maps.
+    """
+    from scipy.ndimage import convolve
+
+    # Kernel is modified later make a copy here
+    kernel = deepcopy(kernel)
+
+    if not kernel.is_bool:
+        log.warn('Using weighted kernels can lead to biased results.')
+
+    kernel.normalize('peak')
+    counts_ = convolve(counts, kernel.array, mode='constant', cval=np.nan)
+    background_ = convolve(background, kernel.array, mode='constant', cval=np.nan)
+
+    significance_lima = significance(counts_, background_, method='lima') 
+
+    result = MapsBunch(significance=significance_lima,
+                       counts=counts_,
+                       background=background_,
+                       excess= counts_ - background_)
+
+    if not exposure is None:
+        kernel.normalize('integral')
+        exposure_ = convolve(exposure, kernel.array, mode='constant', cval=np.nan)
+        flux = (counts_ - background_) / exposure_
+        result.flux = flux
+
+    return result
+
+
+def compute_lima_on_off_map(n_on, n_off, a_on, a_off, kernel, exposure=None,
+                            fft=False):
+    """
+    Compute Li&Ma significance and flux maps for on-off observations.
+
+    Parameters
+    ----------
+    n_on : `~numpy.ndarray`
+        Counts map.
+    n_off : `~numpy.ndarray`
+        Off counts map.
+    a_on : `~numpy.ndarray`
+        Relative .
+    a_off : `~numpy.ndarray`
+        Exposure ratio map.
+    kernel : `astropy.convolution.Kernel2D`
+        convolution kernel. 
+    exposure : `~numpy.ndarray`
+        Exposure map.
+    fft : bool
+        Use fast fft convolution. Default is False.
+    
+    Returns
+    -------
+    Bunch : `gammapy.extern.bunch.Bunch`
+        Bunch of result maps.   
+    """
+    from scipy.ndimage import convolve
+
+    # Kernel is modified later make a copy here
+    kernel = deepcopy(kernel)
+
+    if not kernel.is_bool:
+        log.warn('Using weighted kernels can lead to biased results.')
+
+    kernel.normalize('peak')
+    n_on_ = convolve(n_on, kernel.array, mode='constant', cval=np.nan)
+    a_ = convolve(a_on, kernel.array, mode='constant', cval=np.nan)
+    alpha = a_ / a_off
+    background = alpha * n_off
+
+    significance_lima = significance_on_off(n_on_, n_off, alpha, method='lima')
+
+    # safe significance threshold. Is this worse making an option to
+    # significance()?
+    significance_lima[n_on_ < 5] = 0
+    
+    result = MapsBunch(significance=significance_lima,
+                       n_on=n_on_,
+                       background=background,
+                       excess=n_on_ - background,
+                       alpha=alpha)
+
+    if not exposure is None:
```

Comment:
I just checked,  this doesn't work, when `exposure` is a numpy array. This results in `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
`